src/main/java/fi/fmi/avi/archiver/spring/integration/util/MonitorableCallerBlocksPolicy.java

```diff
@@ -0,0 +1,61 @@
+package fi.fmi.avi.archiver.spring.integration.util;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * {@link RejectedExecutionHandler} similar to {@link org.springframework.integration.util.CallerBlocksPolicy} that
+ * also allows querying the blocked duration.
+ */
+public class MonitorableCallerBlocksPolicy implements RejectedExecutionHandler {
+
+    private final Clock clock;
+    private final long maxWait;
+    private final AtomicLong timestamp = new AtomicLong(-1);
```

Comment:
Name `blockStartMillis` or something like that might express the intention clearer?

Reply:
👍